src/classes/PassageError/PassageError.ts

```diff
@@ -0,0 +1,31 @@
+import { ResponseError } from '../../generated';
+import { PassageErrorCode } from './types';
+/**
+ * PassageError Class used to handle errors from PassageFlex
+ */
+export class PassageError extends Error {
+    override name: 'PassageError' = 'PassageError';
+
+    /**
+     * Initialize a new PassageError instance.
+     * @param {number} statusCode status code from PassageFlex API
+     * @param {string} errorCode error code from PassageFlex API
+     * @param {string} message friendly message
+     */
+    private constructor(
+        public readonly statusCode: number,
+        public readonly errorCode: PassageErrorCode,
+        public readonly message: string,
+    ) {
+        super(message);
+    }
+    /**
+     * Initialize a new PassageError instance.
+     * @param {ResponseError} err error from node-fetch request
+     * @return {Promise<PassageError>}
+     */
+    public static async fromResponseError(err: ResponseError): Promise<PassageError> {
+        const body: { code: PassageErrorCode; error: string } = await err.response.json();
```

Comment:
In theory this operation should not fail, but I suppose if it did what would we want to populate the `errorCode` and `message` fields with?

Reply:
i think the only reason this would fail is because the response body was not parsable as json. perhaps just a brief message and code conveying that we got an unknown response?

i was trying to think if there was something more helpful or actionable we could say, but i don't think parsing the response as raw text and returning that would be helpful.